spring-integration-debezium/src/main/java/org/springframework/integration/debezium/inbound/DebeziumMessageProducer.java

```diff
@@ -0,0 +1,283 @@
+	 */
+	public DebeziumMessageProducer(Builder<ChangeEvent<byte[], byte[]>> debeziumBuilder) {
+		Assert.notNull(debeziumBuilder, "The Debezium Engine Builder is null!");
+		this.debeziumEngineBuilder = debeziumBuilder;
+	}
+
+	/**
+	 * Defines if a single or a batch of messages are send downstream.
+	 * @param enableBatch False - sends downstream {@link Message} for every {@link ChangeEvent data change event} read
+	 * from the source database. True - sends the received {@link List} of {@link ChangeEvent}s as a raw {@link Message}
+	 * payload in a single (batch) downstream message. Such payload is not serializable and would required custom
+	 * serialization/deserialization implementation.
+	 */
+	public void setEnableBatch(boolean enableBatch) {
+		this.enableBatch = enableBatch;
+	}
+
+	/**
+	 * Enables support for tombstone (aka delete) messages. On a database row delete, Debezium can send a tombstone
+	 * change event that has the same key as the deleted row and a value of {@link Optional.empty}. This record is a
+	 * marker for downstream processors. It indicates that log compaction can remove all records that have this key.
+	 * When the tombstone functionality is enabled in the Debezium connector configuration you should enable the empty
+	 * payload as well.
+	 * @param enableEmptyPayload True enables the empty payload handling mechanism. False by default.
+	 */
+	public void setEnableEmptyPayload(boolean enableEmptyPayload) {
+		this.enableEmptyPayload = enableEmptyPayload;
+	}
+
+	/**
+	 * Set a {@link ThreadFactory} for Debezium executor. Defaults to the {@link CustomizableThreadFactory} based on a
+	 * {@code debezium-} prefix.
+	 * @param threadFactory the {@link ThreadFactory} to use.
+	 */
+	public void setThreadFactory(ThreadFactory threadFactory) {
+		Assert.notNull(threadFactory, "'threadFactory' must not be null");
+		this.threadFactory = threadFactory;
+	}
+
+	/**
+	 * Outbound message content type. Should be aligned with the {@link SerializationFormat} configured for the
+	 * {@link DebeziumEngine}.
+	 */
+	public void setContentType(String contentType) {
+		Assert.hasText(contentType, "Invalid content type: " + contentType);
+		this.contentType = contentType;
+	}
+
+	/**
+	 * Specifies how to convert Debezium change event headers into Message headers.
+	 * @param headerMapper concrete HeaderMapping implementation.
+	 */
+	public void setHeaderMapper(HeaderMapper<List<Header<Object>>> headerMapper) {
+		Assert.notNull(headerMapper, "'headerMapper' must not be null.");
+		this.headerMapper = headerMapper;
+	}
+
+	/**
+	 * @return Returns current header mapper.
+	 */
+	public HeaderMapper<List<Header<Object>>> getHeaderMapper() {
+		return this.headerMapper;
+	}
+
+	@Override
+	public String getComponentType() {
+		return "debezium:inbound-channel-adapter";
+	}
+
+	@Override
+	protected void onInit() {
+		super.onInit();
+		Assert.notNull(this.headerMapper, "Header mapper can not be null!");
```

Comment:
According to the `setHeaderMapper` logic it cannot be null, so just remove this line from here.